### PR TITLE
Example: `azurerm_virtual_desktop_*`

### PR DESCRIPTION
**Pull request by Jsheerin**
Added examples/desktopvirtualization files to create WVD host pool, application group, workspace and associate application group to workspace.

### CODE REVIEW

**Review by mbfrahry — CHANGES_REQUESTED** [review]
Hey @Jsheerin, thanks for opening this PR! I've gone over it and the idea is there but it doesn't quite follow the style we'd like to have for examples. I've commented on a few areas but I'd suggest also looking into other examples and copying their structure. We'd like to aim for an end to end example of `desktopvirtualization` before we merge it in. [storage-share](https://github.com/terraform-providers/terraform-provider-azurerm/tree/master/examples/storage/storage-share) would be a good example to model this one after.

File: examples/desktopvirtualization/main.tf [hunk 1]
```diff
@@ -0,0 +1,45 @@
+#terraform
+terraform {
+    required_version        = ">=0.13.4"
```
[the remark this answers is not in the document]
Comment (Jsheerin) [reply examples/desktopvirtualization/main.tf]: Thank you @mbfrahry, I understand after reading your feedback. I have made the updates based on your suggestion and modeling after other examples.

File: examples/desktopvirtualization/main.tf [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+#Azure provider
+provider "azurerm" {
+    version                 = "2.31.1"
+    features {}
+    subscription_id         = var.subscriptionID
+}
+
```
[the remark this answers is not in the document]
Comment (Jsheerin) [reply examples/desktopvirtualization/main.tf:9]: Thank you! I included the resource group

File: examples/desktopvirtualization/main.tf [hunk 3]
```diff
@@ -0,0 +1,45 @@
+#terraform
+terraform {
+    required_version        = ">=0.13.4"
+}
+
+#Azure provider
+provider "azurerm" {
+    version                 = "2.31.1"
```
[the remark this answers is not in the document]
Comment (Jsheerin) [reply examples/desktopvirtualization/main.tf]: I have made the updates based on your suggestion and modeling after other examples.

File: examples/desktopvirtualization/variables.tf [hunk 4]
```diff
@@ -0,0 +1,14 @@
+variable "subscriptionID" {
```
[the remark this answers is not in the document]
Comment (Jsheerin) [reply examples/desktopvirtualization/variables.tf]: Changed as you recommended

File: examples/desktopvirtualization/main.tf [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    features {}
+    subscription_id         = var.subscriptionID
```
[the remark this answers is not in the document]
Comment (Jsheerin) [reply examples/desktopvirtualization/main.tf]: Removed. Thank you!

**Review by mbfrahry — APPROVED** [review]
LGTM!

**Pull request merged**